lib/components/Modal/ModalHeader.tsx

```diff
@@ -0,0 +1,23 @@
+import { HtmlHTMLAttributes, PropsWithChildren, } from 'react';
+import clsx from 'clsx';
+
+import { CloseIcon } from './CloseIcon';
+
+type ModalHeaderProps = HtmlHTMLAttributes<HTMLDivElement> & {
+    hideCloseIcon?: boolean;
+    onRequestClose?: VoidFunction;
+    shouldShowBorder?: boolean;
+    hideBorder?: boolean;
+}
+
+export const ModalHeader = ({ children, hideCloseIcon, className, onRequestClose, hideBorder = false, ...rest }: PropsWithChildren<ModalHeaderProps>) => {
```

Comment:
If I'm not mistaken ComponentProps<'div'> already has `children` prop.
```suggestion
export const ModalHeader = ({ children, hideCloseIcon, className, onRequestClose, hideBorder = false, ...rest }: ModalHeaderProps) => {
```